src/docs/arch-block.dox

```diff
@@ -1,9 +1,192 @@
 /*! @arch_page arch-block Block Manager
 
-The Block Manager manages the reading and writing of disk blocks
-in WiredTiger.  It does compression and encryption when these
-are configured.
+The WiredTiger block manager subsystem manages the reading and writing of data
+from the disk. It is designed to facilitate high performance, economic use of
+disk space and customizability.
+
+@section block What is a block?
+
+A block is a chunk of data that is stored on the disk and operated on as a
+single unit. Each WiredTiger data file (any file in the home directory with the
+\c .wt suffix) is made up of these blocks. Each block consists of a page header,
+a block header and contains a single page from the btree which it was
+generated. Blocks can only be written once and a new block is created upon
+reconciliation of a dirty page. The size of a block is a multiple of the
+allocation size which is set during creation of associated WiredTiger data file
+see: WT_SESSION::create.
+
+Once a block is written the address of the block is returned and is referred to
```

Comment:
It would be infinitely useful to explicitly say _"the term 'address cookie' refers to the address returned by the block manager, which is not a literal address but rather ..."_. We have a lot of terms flying around in the code comments so it would be helpful to define what they mean in the documentation.

Reply:
Does the most recent commit achieve what you both expected?

Reply:
Looks good.  I also pinged Don to suggest he include "address cookie" in the glossary.

Reply:
On second thought, it would also be good to point out that the address cookie is opaque to higher levels of the system. Only the block manager understands the contents of the cookie.

Reply:
6cd88e1